tlatools/org.lamport.tlatools/src/tlc2/tool/fp/OffHeapDiskFPSet.java

```diff
@@ -604,13 +604,14 @@ public static class Indexer {
 		
 		public Indexer(final long positions, final int fpBits) {
 			this(positions, fpBits, 0xFFFFFFFFFFFFFFFFL >>> fpBits);
+			Assert.check(positions < Integer.MAX_VALUE, EC.SYSTEM_FINGERPRINT_OVERFLOW_ERROR);
 			assert fpBits > 0;
 		}
 
 		public Indexer(final long positions, final int fpBits, final long maxFingerprint) {
 			this.positions = positions;
 			// (position-1L) because array is zero indexed.
-			this.tblScalingFactor = (positions - 1L) / ((maxFingerprint - minFingerprint) * 1f);
+			this.tblScalingFactor = (positions - 1L) / ((maxFingerprint - minFingerprint) * 1d);
```

Comment:
This is the actual fix.

Reply:
Not all longs are representable as doubles. Is there any chance that this still isn't enough precision?